rts/Lua/LuaHandleSynced.cpp

```diff
@@ -2120,16 +2120,187 @@ int CSyncedLuaHandle::GetWatchWeaponDef(lua_State* L) {
+ * ```lua
+ * Spring.GetWatchExplosion(weaponDefID) or Spring.GetWatchProjectile(weaponDefID) or Spring.GetWatchAllowTarget(weaponDefID)
+ * ```
+ *
+ * @param weaponDefID integer
+ * @return boolean watched True if watch is enabled for any weaponDefID callins.
+ *
+ * @see Spring.SetWatchWeapon
+ */
+
+/*** Query Explosion watch status for a weaponDefID.
+ *
+ * @function Spring.GetWatchExplosion
+ *
+ * @param weaponDefID integer
+ * @return boolean watched Watch status.
+ *
+ * @see Spring.SetWatchExplosion
+ */
+
 GetWatchDef(Explosion)
+
+
+/*** Query Projectile watch status for a weaponDefID.
+ *
+ * @function Spring.GetWatchProjectile
+ *
+ * @param weaponDefID integer
+ * @return boolean watched Watch status.
+ *
+ * @see Spring.SetWatchProjectile
+ */
+
 GetWatchDef(Projectile)
+
+
+/*** Query AllowTarget watch status for a weaponDefID.
+ *
+ * @function Spring.GetWatchAllowTarget
+ *
+ * @param weaponDefID integer
+ * @return boolean watched Watch status.
+ *
+ * @see Spring.SetWatchAllowTarget
+ */
+
 GetWatchDef(AllowTarget)
 
+
+/*** Register or deregister unitDefID for expensive callins.
+ *
+ * @function Spring.SetWatchUnit
+ *
+ * @param unitDefID integer
+ * @param watch boolean Whether to register or deregister.
+ * @return nil
+ *
+ * @see Callins:UnitFeatureCollision
+ * @see Callins:UnitUnitCollision
+ * @see Callins:UnitMoveFailed
+ */
+
 SetWatchDef(Unit)
+
+
+/*** Register or deregister featureDefID for expensive callins.
+ *
+ * @function Spring.SetWatchFeature
+ *
+ * @param featureDefID integer
+ * @param watch boolean Whether to register or deregister.
+ * @return nil
+ *
+ * @see Callins:UnitFeatureCollision
+ */
+
 SetWatchDef(Feature)
+
+
+/*** Register or deregister weaponDefID for all expensive callins.
+ *
+ * @function Spring.SetWatchWeapon
+ *
+ * Equivalent to calling:
+ *
+ * ```lua
+ * Spring.SetWatchExplosion(weaponDefID)
+ * Spring.SetWatchProjectile(weaponDefID)
+ * Spring.SetWatchAllowTarget(weaponDefID)
+ * ```
+ *
+ * Generally it's better to use those methods to avoid registering uneeded callins.
+ *
+ * @param weaponDefID integer
+ * @param watch boolean Whether to register or deregister.
+ * @return nil
+ */
+
+/*** Register or deregister weaponDefID for expensive Explosion callins.
+ *
+ * @function Spring.SetWatchExplosion
+ *
+ * @param weaponDefID integer
+ * @param watch boolean Whether to register or deregister.
+ * @return nil
+ *
+ * @see SyncedCallins:Explosion
+ */
+
 SetWatchDef(Explosion)
+
+
+/*** Register or deregister weaponDefID for expensive Projectile callins.
+ *
+ * @function Spring.SetWatchProjectile
+ *
+ * @param weaponDefID integer weaponDefID for weapons or -1 to watch for debris.
+ * @param watch boolean Whether to register or deregister.
+ * @return nil
+ *
+ * @see SyncedCallins:ProjectileCreated
+ * @see SyncedCallins:ProjectileDestroyed
+ */
+
 SetWatchDef(Projectile)
+
+
+/*** Register or deregister weaponDefID for expensive AllowWeaponTarget callins.
```

Comment:
```suggestion
/*** Register or deregister weaponDefID for weapon targeting callins.
```

PascalCase makes it read like `AllowWeaponTarget` is the only callin, but the `@see` section suggests these are three different callins with other names.

Alternatively list each callin by name.

If the callins are expensive this probably should be mentioned as an additional line like "These callins are disabled by default because they are expensive". Although I'd suggest this might be better to add to the callin itself e.g.

> This callin is disabled by default because it is expensive, to enable the callin use `SetWatchAllowTarget`.